pkg/service/session_manager.go

```diff
@@ -64,23 +67,25 @@ func (sm *SessionManager) IngressStarted(info *livekit.IngressInfo, sessionAPI t
 
 	sm.sessions[info.State.ResourceId] = r
 
+	sm.registerKillIngressSession(info.IngressId, info.State.ResourceId)
+
 	sm.monitor.IngressStarted(info)
 }
 
-func (sm *SessionManager) IngressEnded(info *livekit.IngressInfo) {
-	logger.Infow("ingress ended", "ingressID", info.IngressId, "resourceID", info.State.ResourceId)
-
+func (sm *SessionManager) IngressEnded(resourceID string) {
 	sm.lock.Lock()
 	defer sm.lock.Unlock()
 
-	p := sm.sessions[info.State.ResourceId]
+	p := sm.sessions[resourceID]
```

Comment:
If I understood correctly, `ListIngress` may return an empty `resourceID`, which will be skipped here. Not sure how frequent that is. Is it worth checking before locking the mutex?

Reply:
We need to deploy ingress before cloud-io. Nobody will call KillIIngressSession before we do so.